src/main/java/com/wultra/app/mobileutilityserver/config/SecurityConfig.java

```diff
@@ -0,0 +1,117 @@
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true, jsr250Enabled = true)
+public class SecurityConfig {
+
+    @Value("${mobile-utility-server.security.auth.basicHttp.stateless}")
+    private boolean stateless;
+
+    @Value("${mobile-utility-server.security.auth.algorithm}")
+    private String algorithm;
+
+    @Value("${mobile-utility-server.security.auth.bcrypt.cycles}")
+    private int bcryptCycles;
+
+    private final String[] supportedHashAlgorithms = { "SHA-256", "bcrypt" };
+
+    @Bean
+    public UserDetailsService userDetailsService(PasswordEncoder passwordEncoder) {
+        final InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
+        manager.createUser(User.withUsername("admin")
+                .password(passwordEncoder.encode("admin"))
+                .roles("ADMIN")
+                .build());
+        return manager;
+    }
+
+    /**
+     * Configure SHA-256 or bcrypt password encoder. Note that since the passwords are technical, using old SHA-256
+     * algorithm does not cause security issues. Bcrypt is used as default in case no prefix is specified.
+     * See the following URL for constant details:
+     * <a href="https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/crypto/factory/PasswordEncoderFactories.html#createDelegatingPasswordEncoder()">PasswordEncoderFactories.createDelegatingPasswordEncoder()</a>
+     * @return Delegating password encoder.
+     */
+    @Bean
+    @SuppressWarnings("deprecation")
+    public PasswordEncoder passwordEncoder() throws NoSuchAlgorithmException {
+        if (!List.of(supportedHashAlgorithms).contains(algorithm)) {
+            throw new NoSuchAlgorithmException(String.format("Unsupported algorithm specified: %s, must be one of: %s.", algorithm, Arrays.toString(supportedHashAlgorithms)));
+        }
+        final BCryptPasswordEncoder bcrypt = new BCryptPasswordEncoder(bcryptCycles);
+        final MessageDigestPasswordEncoder sha256 = new MessageDigestPasswordEncoder("SHA-256");
+        final Map<String, PasswordEncoder> encoders = new HashMap<>();
+        encoders.put("bcrypt", bcrypt);
+        encoders.put("SHA-256", sha256);
+        final DelegatingPasswordEncoder passwordEncoder = new DelegatingPasswordEncoder(algorithm, encoders);
+        passwordEncoder.setDefaultPasswordEncoderForMatches(bcrypt); // try using bcrypt as default, for backward compatibility
+        return passwordEncoder;
+    }
+
+    /**
+     * Configure security filter chain.
+     * @param http HTTP configuration.
+     * @return Security filter chain.
+     * @throws Exception In case a configuration error occurs.
+     */
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http
+            .csrf().disable()
+            .authorizeRequests()
+                .antMatchers("/admin/**").hasRole("ADMIN")
+                .anyRequest().permitAll()
```

Comment:
We may drop the usage of `.and()` completely and switch to lambda style
```suggestion
            .authorizeRequests(authorize -> authorize
                .antMatchers("/admin/**").hasRole("ADMIN")
                .anyRequest().permitAll())
```